<?xml version="1.0"?>
<launch>

  <!-- Camera ID -->
  <arg name="camera_id"               default="sr4500"/>
  <arg name="device_ip"               default="192.168.1.33" />
  <arg name="ground_from_calibration" default="false" />
  <arg name="camera_info_url"         default="file://$(env HOME)/.ros/camera_info/$(arg camera_id).yaml" />

  <!-- Start the SR4500 -->
  <include file="$(find swissranger_camera)/launch/sr_eth.launch">
    <arg name="camera_id"       value="$(arg camera_id)" />
    <arg name="device_ip"       value="$(arg device_ip)" />
    <arg name="camera_info_url" value="$(arg camera_info_url)" />
  </include>

  <!-- Launch ground based people detection node -->
  <node pkg="detection" type="ground_based_people_detector_sr"
        name="ground_based_people_detector_$(arg camera_id)" output="screen" respawn="true">
    
    <rosparam command="load" file="$(find detection)/conf/ground_based_people_detector_sr_multicamera.yaml" /> 
    
    <param name="classifier_file"                   value="$(find detection)/data/HogSvmPCL.yaml"/>
    <param name="pointcloud_topic"                  value="/$(arg camera_id)/pointcloud2_raw"/>
    <param name="confidence_topic"                  value="/$(arg camera_id)/confidence/image_raw"/>
    <param name="output_topic"                      value="/detector/detections"/>
    <param name="camera_info_topic"                 value="/$(arg camera_id)/camera_info"/>
    <param name="rate"                              value="60.0"/>  
    <param name="ground_from_extrinsic_calibration" value="$(arg ground_from_calibration)"/>  
    
  </node>

</launch>
